token/core/fabtoken/validator.go

```diff
@@ -82,6 +154,20 @@ func (v *Validator) VerifyTokenRequest(ledger driver.Ledger, signatureProvider d
 	for _, action := range ta {
 		actions = append(actions, action)
 	}
+	for _, sig := range signatureProvider.Signatures() {
+		claim := &exchange.ClaimSignature{}
+		if err = json.Unmarshal(sig, claim); err != nil {
+			continue
+		}
+		if len(claim.Preimage) == 0 || len(claim.RecipientSignature) == 0 {
```

Comment:
Shouldn't this be in-acceptable. If the unmarshalling is successful, shouldn't `Pre-image` and  `RecipientSignature` not be empty.